### PR TITLE
(MODULES-7100) Update Readme documentation

### PR DESCRIPTION
**Pull request by willmeek**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,6 +112,20 @@ Declaring these resources will configure `device.conf` and apply the base class
...
+#### Signing certificates on first run
+
+If this is the first time executing against this device and the certificate has not yet been signed, the following may be seen:
+```bash
+Notice: Did not receive certificate
+```
+
+Execute the following to sign the certificates for the devices:
+```bash
+puppet cert sign --all
```
Comment (tkishel) [inline README.md]: `puppet cert sign --all` is unsafe, since any other pending certificate requests will be signed.

Comment (willmeek) [reply README.md]: Good point.

I've updated with just the hostname, or will I remove this section entirely and assume the target audience has enough knowledge of Puppet?

Comment (tkishel) [reply README.md]: I would think a seasoned user would know, but who's to say a newer user would explore with a device? Seems good with a hostname.

File: README.md [hunk 2]
```diff
@@ -112,6 +112,20 @@ Declaring these resources will configure `device.conf` and apply the base class
...
+#### Signing certificates on first run
+
+If this is the first time executing against this device and the certificate has not yet been signed, the following may be seen:
+```bash
+Notice: Did not receive certificate
+```
+
+Execute the following to sign the certificates for the device(s):
+```bash
+puppet cert sign 'bigip1.example.com'
```
Comment (tkishel) [inline README.md]: Maybe change to 'bigip.example.com' to match preceding hostname reference?

File: README.md [hunk 3]
```diff
@@ -134,19 +148,21 @@ Setting to 'absent' deletes the device from `device.conf` and the `devices` fact
 
 Data type: String
 
-Specifies the type of the device in `device.conf` on the proxy Puppet agent.
+Specifies the type of the device in `device.conf` on the proxy Puppet agent. This is the module type to be used to access the device.
```
Comment (clairecadman) [inline README.md]: Omit "to be"

File: README.md [hunk 4]
```diff
@@ -134,19 +148,21 @@ Setting to 'absent' deletes the device from `device.conf` and the `devices` fact
...
 
 ### url
 
 Data type: String
 
+This parameter is optional - either credentials or url must be used, however these are mutually exclusive (only one must be used).
+
```
Comment (clairecadman) [inline README.md]: This is a bit confusing, isn't the second part of the sentence saying the same thing as the first half?

Switch to active voice "use either the credentials or url".

Also should it be url or URL?

Comment (willmeek) [reply README.md]: Changed. I've kept the parameter title in lowercase, but reference to it as uppercase?

"This parameter is optional - use either the credentials or URL." - for url

"This parameter is optional and specific to devices that use the Puppet Resource API. Use either the credentials or URL." - for credentials. I kept in the existing note about Resource API.

Thanks!

**Comment by willmeek** [comment]
If we feel that the comments have been addressed and happy with how the readme looks?

**Comment by tkishel** [comment]
I don't want to slow things down, but I'm composing a comment or two right now ...

**Comment by tphoney** [comment]
It might be best to edit the pr inline.  editing in the github ui

**Pull request merged**